whd/dispatcher.py

```diff
@@ -318,6 +318,56 @@ def is_up_to_date(resource, resource_versions):
             sleep_seconds=sleep_seconds*1.2,
         )
 
+    def _jobs_not_triggered(self, concourse_api, matching_resources, pr_event):
+        '''
+        return all jobs and the corresponding resource versions, which have not been triggered
+        '''
+        for resource in matching_resources:
+            pipeline_name = resource.pipeline.name
+            resource_versions = concourse_api.resource_versions(
+                pipeline_name=pipeline_name,
+                resource_name=resource.name
+            )
+            pr_resource_versions = [
+                rv for rv in resource_versions if rv.version()['pr'] == str(pr_event.number())
+            ]
+            if not pr_resource_versions:
+                logger.warning(
+                    f"no resource versions found for pipeline {resource.pipeline.name} "
+                    f"resource {resource.name} and PR {pr_event.number()}"
+                )
+                return
+
+            # get all jobs which get triggered by the resource
+            jobs_triggered_by_resource = [
```

Comment:
it is the jobs that _should_ be triggered by the resource. so please reflect this in the name (and comment!)

Reply:
done